src/features/defi/providers/foxy/components/FoxyManager/Overview/FoxyOverview.tsx

```diff
@@ -60,44 +65,57 @@ export const FoxyOverview: React.FC<FoxyOverviewProps> = ({
-    [accountId, highestBalanceAccountId],
-  )
-  const bip44Params = useAppSelector(state => selectBIP44ParamsByAccountId(state, accountFilter))
-  const { data: foxyBalancesData, isLoading: isFoxyBalancesLoading } = useFoxyBalances({
-    accountNumber: bip44Params?.accountNumber ?? 0,
-  })
   const translate = useTranslate()
 
-  const opportunity = useMemo(
-    () => (foxyBalancesData?.opportunities || []).find(e => e.contractAssetId === assetId),
-    [foxyBalancesData?.opportunities, assetId],
+  const defaultAccountId = useAppSelector(state => selectFirstAccountIdByChainId(state, chainId))
+  const maybeAccountId = useMemo(
+    () => accountId ?? highestBalanceAccountId ?? defaultAccountId,
+    [accountId, defaultAccountId, highestBalanceAccountId],
   )
+  useEffect(() => {
+    if (!maybeAccountId) return
+    handleAccountIdChange(maybeAccountId)
+  }, [handleAccountIdChange, maybeAccountId])
 
-  const withdrawInfo = accountId
-    ? // Look up the withdrawInfo for the current account, if we have one
-      opportunity?.withdrawInfo[accountId]
-    : // Else, get the withdrawInfo for the highest balance account
-      opportunity?.withdrawInfo[highestBalanceAccountId ?? '']
-  const rewardBalance = bnOrZero(withdrawInfo?.amount)
-  const releaseTime = withdrawInfo?.releaseTime
-  const foxyBalance = bnOrZero(opportunity?.balance)
+  const opportunityDataFilter = useMemo(() => {
+    return {
+      userStakingId: serializeUserStakingId(
+        accountId ?? highestBalanceAccountId ?? '',
+        assetId as StakingId,
+      ),
+    }
+  }, [accountId, assetId, highestBalanceAccountId])
 
-  const marketData = useAppSelector(state => selectMarketDataById(state, stakingAssetId))
-  const cryptoAmountAvailablePrecision = bnOrZero(foxyBalance).div(
-    bn(10).pow(stakingAsset?.precision ?? 0),
+  const foxyEarnOpportunityData = useAppSelector(state =>
+    opportunityDataFilter
+      ? selectEarnUserStakingOpportunityByUserStakingId(state, opportunityDataFilter)
+      : undefined,
+  )
+
+  const hasClaim = useAppSelector(state =>
+    opportunityDataFilter ? selectHasClaimByUserStakingId(state, opportunityDataFilter) : undefined,
+  )
+
+  const undelegation = useMemo(
+    () =>
+      foxyEarnOpportunityData && supportsUndelegations(foxyEarnOpportunityData)
+        ? foxyEarnOpportunityData.undelegations[0]
+        : undefined,
+    [foxyEarnOpportunityData],
   )
+
+  const marketData = useAppSelector(state => selectMarketDataById(state, stakingAssetId))
+  const cryptoAmountAvailablePrecision = bnOrZero(
+    foxyEarnOpportunityData?.stakedAmountCryptoBaseUnit,
+  ).div(bn(10).pow(stakingAsset?.precision ?? 0))
   const fiatAmountAvailable = bnOrZero(cryptoAmountAvailablePrecision).times(marketData.price)
-  const claimAvailable = dayjs().isAfter(dayjs(releaseTime))
-  const hasClaim = rewardBalance.gt(0)
+  const claimAvailable = dayjs().isAfter(dayjs('0')) // TODO
```

Comment:
is this a todo like i need to fix in this PR, or a todo like it's finished up next in the stack?

Reply:
Definitely a legit TODO: 86a863f95ba1a4f4e6e162919655c870c5148f6d